base_layer/wallet/src/output_manager_service/storage/sqlite_db.rs

```diff
@@ -114,23 +147,51 @@ impl OutputManagerBackend for OutputManagerSqliteDatabase {
                     None
                 },
             },
-            DbKey::UnspentOutputs => Some(DbValue::UnspentOutputs(
-                OutputSql::index_status(OutputStatus::Unspent, &(*conn))?
-                    .iter()
-                    .map(|o| DbUnblindedOutput::try_from(o.clone()))
-                    .collect::<Result<Vec<_>, _>>()?,
-            )),
-            DbKey::SpentOutputs => Some(DbValue::SpentOutputs(
-                OutputSql::index_status(OutputStatus::Spent, &(*conn))?
-                    .iter()
-                    .map(|o| DbUnblindedOutput::try_from(o.clone()))
-                    .collect::<Result<Vec<_>, _>>()?,
-            )),
+            DbKey::UnspentOutputs => {
+                let mut outputs = OutputSql::index_status(OutputStatus::Unspent, &(*conn))?;
+
+                if let Some(cipher) = (*cipher).clone() {
+                    for o in outputs.iter_mut() {
+                        o.decrypt(&cipher)
+                            .map_err(|_| OutputManagerStorageError::AeadError("Decryption Error".to_string()))?;
+                    }
+                }
```

Comment:
This should work (same for many other places) 
```suggestion
                if let Some(cipher) = cipher.as_ref() {
                    for o in outputs.iter_mut() {
                        o.decrypt(cipher)
                            .map_err(|_| OutputManagerStorageError::AeadError("Decryption Error".to_string()))?;
                    }
                }
```

Reply:
interesting. TIL